### ISSUE TITLE
Possibly neater and simpler way of handling remotes

### ISSUE THREAD

**Issue by ulfalizer** [zephyrproject-rtos/west#83]
One thing I had somehow managed to miss all along is that Git allows URLs wherever names of remotes are allowed. You can do `git fetch https://github.com/zephyrproject-rtos/zephyr.git`, for example. That feature might even predate remotes, which are more like bookmarks (with some associated configuration settings, like where to store the refs that are fetched).

By always fetching directly from the URL, we could bypass the entire remote system. The nice thing about that is that we no longer rely on how remotes are configured. It's also really close to what @carlescufi wanted to see, only I thought it'd be a lot messier to implement due to remote management.

Another neat thing is that both of the following will just work (maybe they already do), with no extra work required:

```
[west]
remote = my-custom-remote
revision = master
```

```
[west]
remote = https://github.com/zephyrproject-rtos/west
revision = master
```

Any remotes we create would simply be for the user's convenience. They could even delete them or point them somewhere else, and things would still work.

**Comment by ulfalizer** [comment]
This came up while looking at https://github.com/zephyrproject-rtos/west/issues/77 by the way. I'll see how well this works first, as it would help with that issue.

**Comment by carlescufi** [comment]
That is a really good proposal and will solve a lot of issues that might arise from users that do not know how to manage Git remotes.
+1 to this.

**Comment by carlescufi** [comment]
CC @mbolivar @tejlmand

**Comment by tejlmand** [comment]
Agree, this looks very promising.
+1.

**Comment by carlescufi** [comment]
> I also really like this idea. I don't understand what "bypass the entire remote system" means. Surely you aren't suggesting removal of that concept from the manifest schema?

No, I think @ulfalizer means not  using Git's indirection of `remote` -> `url`

**Comment by mbolivar** [comment]
> No, I think @ulfalizer means not using Git's indirection of `remote` -> url

If that's the case that sounds great.

**Comment by ulfalizer** [comment]
Yeah, it's just switching to specifying repositories via URLs instead of Git remotes. That removes remote management and gives less ways for things to break.

After the change, we never use Git remotes ourselves, except we create one for the user's sake.
